linkerd/app/core/src/retry.rs

```diff
@@ -138,6 +139,12 @@ where
         *clone.headers_mut() = req.headers().clone();
         *clone.version_mut() = req.version();
 
+        // If the ClientHandle extension is present, clone it into the new
+        // request.
```

Comment:
i would prefer the comment to explain _why_ we are doing this, rather than _what_ we are doing ---  "if the client handle is present, clone it into the new request" is actually pretty obvious from the code, but something like this is less obvious:
```suggestion
        // The cloned request may also need the ability to tear down the
        // client connection if the remote proxy returns a `l5d-proxy-error`
        // header, so ensure it also has a `ClientHandle`.
        // If the ClientHandle extension is present, clone it into the new
        // request.
```

Reply:
Great point